.github/actions/send-telegram-notify/action.yml

```diff
@@ -0,0 +1,30 @@
+name: 'Send Telegram message on failure'
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        python -m pip install --upgrade pip
+        pip install requests
+      shell: bash
+    # Let's logout github environment
+    - env:
+        GITHUB_CONTEXT: ${{ toJson(github) }}
+      run: echo "$GITHUB_CONTEXT"
+      shell: bash
+    - env:
+        MSG: >-
+          🔴 Workflow testing failed:\n
+          Job: [ `${{ github.job }}`](https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }})\n
+          Commit: [ `${{ github.sha }}`](https://github.com/${{ github.repository }}/commit/${{ github.sha }})\n
+          Branch: [ `${{ env.CI_ACTION_REF_NAME }}`](https://github.com/${{ github.repository }}/tree/${{ github.ref }})\n
+          History: [commits](https://github.com/${{ github.repository }}/commits/${{ github.sha }})\n
+          Triggered on: `${{ github.event_name }}`\n
+          Committer: `${{ github.actor }}`
+          ```
+          -------------------- Commit message --------------------------\n
+          ${{ github.event.head_commit.message }}\n
+          ```
+      # Use MarkdownV2 while Markdown is legacy
+      # https://core.telegram.org/bots/api#markdownv2-style
+      run: python -c "import urllib, requests ; url = 'https://api.telegram.org/bot%s/sendMessage?chat_id=%s&text=%s&parse_mode=MarkdownV2' % ('$TELEGRAM_TOKEN', '$TELEGRAM_TO', urllib.quote_plus('$MSG')) ; _ = requests.get(url, timeout=10)"
```

Comment:
I would prefer using POST (Telegram [supports](https://core.telegram.org/bots/api#making-requests) GET and POST both). There are two reasons.

1. Theoretical. GET is usually used for receiving data, not for sending it.
2. Practical. GET may be cached by, say, a proxy server.

Aside of this, `requests` supports automatic urlencoding, just pass parameters like so:

```python
params = {
    'char_id': <...>,
    'text': <...>,
    'parse_mode': <...>,
}
requests.get(url, params=params)
```

If you'll use POST, than, well, `requests` supports it too:

```python
requests.post(url, data=<...>)
```

Or, if you prefer to send JSON:

```python
requests.post(url, json=<...>)
```

Both ways should handle escaping for you.